assets/js/controllers/settings/info.controller.js

```diff
@@ -2,30 +2,29 @@ angular
   .module('walletApp')
   .controller('SettingsInfoCtrl', SettingsInfoCtrl);
 
-function SettingsInfoCtrl ($scope, Wallet, Alerts, $translate, $window) {
-  $scope.uid = Wallet.user.uid;
+function SettingsInfoCtrl ($scope, $q, Wallet, Alerts) {
+  angular.extend($scope, Wallet.user);
+  $scope.loading = {};
+  $scope.pairingCode = null;
 
-  $scope.display = {
-    pairingCode: false
+  $scope.removeAlias = () => {
+    $scope.loading.alias = true;
+    Alerts.confirm('CONFIRM_DISABLE_ALIAS', { id: $scope.guid })
+      .then(Wallet.removeAlias)
+      .then(() => $scope.alias = Wallet.user.alias, () => {})
+      .then(() => $scope.loading.alias = false);
   };
-  $scope.pairingCode = null;
 
   $scope.hidePairingCode = () => {
     $scope.pairingCode = null;
-    $scope.display.pairingCode = false;
   };
 
   $scope.showPairingCode = () => {
-    const success = (pairingCode) => {
-      $scope.pairingCode = pairingCode;
-      $scope.loading = false;
-      $scope.display.pairingCode = true;
-    };
-    const error = () => {
-      Alerts.displayError('Failed to load pairing code.');
-      $scope.loading = false;
-    };
-    $scope.loading = true;
-    Wallet.makePairingCode(success, error);
+    $scope.loading.code = true;
+    let success = (code) => $scope.pairingCode = code;
+    let error = () => Alerts.displayError('SHOW_PAIRING_CODE_FAIL');
+    $q(Wallet.makePairingCode)
+      .then(success, error)
+      .then(() => $scope.loading.code = false);
```

Comment:
Yeah, because the prior `.then` will handle any errors. So that last `.then` will always run.